src/Application/Messenger/Handler/MandateUpsertedHandler.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace MatchBot\Application\Messenger\Handler;
+
+use Doctrine\ORM\EntityManagerInterface;
+use MatchBot\Application\Assertion;
+use MatchBot\Application\Messenger\AbstractStateChanged;
+use MatchBot\Application\Messenger\DonationUpserted;
+use MatchBot\Application\Messenger\MandateUpserted;
+use MatchBot\Client\Mandate;
+use MatchBot\Domain\DoctrineDonationRepository;
+use MatchBot\Domain\DonationRepository;
+use MatchBot\Domain\Salesforce18Id;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Clock\Clock;
+use Symfony\Component\Messenger\Attribute\AsMessageHandler;
+
+/**
+ * Sends donations to Salesforce.
+ */
+#[AsMessageHandler]
+readonly class MandateUpsertedHandler
+{
+    /**
+     * @psalm-suppress PossiblyUnusedMethod
+     */
+    public function __construct(
+        private LoggerInterface $logger,
+        private Mandate $client,
+        private EntityManagerInterface $entityManager,
+        private Clock $clock,
+    ) {
+    }
+
+    public function __invoke(MandateUpserted $message): void
+    {
+        $this->logger->info("MUH invoked for UUID: $message->uuid");
+        $sfId = $this->client->createOrUpdate($message);
+        $this->setSalesforceFields($message->uuid, $sfId);
```

Comment:
surprisingly to me this already seems to be working on my local - the mandates are getting SF ID's set in my local DB. I thought we would still need to create the endpoint in SF to accept the mandate upserts.

Logging in to full isn't working for me now though so I can't actually see how how the mandates look there.

Reply:
realised it wasn't working,  was just getting confused by seeing artifacts of automated testing in my local db